pkg/iniparser_test.go

```diff
@@ -0,0 +1,1308 @@
+package iniparser
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type LoadFromStringTestCase struct {
+	desc          string
+	input         string
+	expectedError error
+	expectedMap   map[string]map[string]string
+}
+
+type LoadFromFileTestCase struct {
+	desc          string
+	path          string
+	expectedError error
+	expectedMap   map[string]map[string]string
+}
+
+type StringTestCase struct {
+	desc     string
+	inputMap map[string]map[string]string
+}
+
+type SaveToFileTestCase struct {
+	desc          string
+	inputPath     string
+	inputMap      map[string]map[string]string
+	expectedError error
+}
+
+type GetSectionNamesTestCase struct {
+	desc        string
+	inputMap    map[string]map[string]string
+	outputSlice []string
+}
+
+type GetSectionsTestCase struct {
+	desc           string
+	inputOutputMap map[string]map[string]string
+}
+
+type GetSetTestCase struct {
+	desc          string
+	inputMap      map[string]map[string]string
+	section       string
+	key           string
+	value         string
+	expectedError error
+}
+
+func sliceEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	a_copy := make([]string, len(a))
+	b_copy := make([]string, len(b))
+
+	copy(a_copy, a)
+	copy(b_copy, b)
+
+	sort.Strings(a_copy)
+	sort.Strings(b_copy)
+
+	return reflect.DeepEqual(a_copy, b_copy)
+}
+
+func TestINI_LoadFromString(t *testing.T) {
+	parser := IniFile{}
+	emptyMap := make(map[string]map[string]string)
+	testCases := []LoadFromStringTestCase{
+		{
+			desc:          "Empty string as input",
+			input:         "",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one comment line start with # as input",
+			input:         "#This is a comment",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one comment line start with ; as input",
+			input:         ";This is a comment",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one key value line as input",
+			input:         "key = value",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one section line as input",
+			input:         "[section]",
+			expectedError: nil,
+			expectedMap:   map[string]map[string]string{"section": make(map[string]string)},
+		},
+		{
+			desc: "Only comments as input",
+			input: "!This is a comment 1\n " +
+				"!This is a comment 2\n" +
+				"!This is a comment 3\n" +
+				"!This is a comment 4",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
```

Comment:
you're adding unnecessarily too much test cases that are not different from each other, why adding 3 test cases just for comments?